src/azul/indexer/document.py

```diff
@@ -361,6 +363,24 @@ class VersionType(Enum):
 C = TypeVar('C', bound=DocumentCoordinates)
 
 
+class SimpleSourceRef(SourceRef[SimpleSourceName, 'SimpleSourceRef']):
+
+    @classmethod
+    def from_aggregate_source(cls, source):
+        return cls(id=one(source['source'])['id'],
+                   name=SimpleSourceName(one(source['source'])['name']))
```

Comment:
… and this `one(source['source'])['name'])` is an implementation detail of Document (or subclass) specific to the call site.